### PR TITLE
Enable PostHog for page views

### PR DESCRIPTION
**Pull request by jtbandes**
### Changelog
None

### Description
Supersedes / closes #124.

We cannot use the [posthog-docusaurus](https://github.com/PostHog/posthog-docusaurus) plugin because it doesn't allow overriding the `sanitize_properties` function (the custom options passed in go through JSON.stringify twice so a function cannot be included).

This PR adds a new plugin (in the repo) to initialize posthog and log pageviews.

### CODE REVIEW

**Comment by jtbandes** [comment]
Something weird is happening with `sanitize_properties` where it gets stripped out during the build. We may need to revive #186.

**Comment by defunctzombie** [comment]
`sanitize_properties` seems like a strange approach to adding properties. Is this a trick that posthog recommends? Is there a configuration that allows adding custom properties that does not require this function?

**Comment by jtbandes** [comment]
> `sanitize_properties` seems like a strange approach to adding properties. Is this a trick that posthog recommends? Is there a configuration that allows adding custom properties that does not require this function?

This is how we do it in our other apps. I googled and did not find anything. Did you find some alternative?

File: packages/plugin-posthog-pageviews/module.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+if (ExecutionEnvironment.canUseDOM) {
+  console.log("Initializing PostHog");
+  posthogJs.init(FOXGLOVE_PLUGIN_POSTHOG_PAGEVIEWS_API_KEY, {
+    api_host: "https://e.foxglove.dev",
+    // localStorage+cookie is required on web to support cross-subdomain tracking
+    persistence: "localStorage+cookie",
+
+    sanitize_properties: (properties) => {
+      return { ...properties, Platform: "Web", Source: "Docs" };
```
Comment (defunctzombie) [inline packages/plugin-posthog-pageviews/module.ts:15]: Do we really need `web` platform? I'd imagine posthog has something for that already?

**Review by defunctzombie — APPROVED** [review]

File: packages/plugin-posthog-pageviews/module.ts (hunk 1 above)
Comment (jtbandes) [reply packages/plugin-posthog-pageviews/module.ts:15]: I just do what I'm told

**Pull request merged**